src/region_plane.cpp

```diff
@@ -61,15 +65,41 @@ RegPlane::RegPlane(LAMMPS *lmp, int narg, char **arg) :
     zstyle = CONSTANT;
   }
 
+  if (utils::strmatch(arg[5], "^v_")) {
+    nxstr = utils::strdup(arg[5] + 2);
+    normal[0] = 0.0;
+    nxstyle = VARIABLE;
+    varshape = 1;
+  } else {
+    normal[0] = xscale * utils::numeric(FLERR, arg[5], false, lmp);
+    nxstyle = CONSTANT;
+  }
+
+  if (utils::strmatch(arg[6], "^v_")) {
+    nystr = utils::strdup(arg[6] + 2);
+    normal[1] = 0.0;
+    nystyle = VARIABLE;
+    varshape = 1;
+  } else {
+    normal[1] = yscale * utils::numeric(FLERR, arg[6], false, lmp);
+    nystyle = CONSTANT;
+  }
+
+  if (utils::strmatch(arg[7], "^v_")) {
+    nzstr = utils::strdup(arg[7] + 2);
+    normal[2] = 0.0;
+    nzstyle = VARIABLE;
+    varshape = 1;
+  } else {
+    normal[2] = zscale * utils::numeric(FLERR, arg[7], false, lmp);
+    nzstyle = CONSTANT;
+  }
+
   if (varshape) {
     variable_check();
     RegPlane::shape_update();
   }
 
-  normal[0] = xscale * utils::numeric(FLERR, arg[5], false, lmp);
-  normal[1] = yscale * utils::numeric(FLERR, arg[6], false, lmp);
-  normal[2] = zscale * utils::numeric(FLERR, arg[7], false, lmp);
-
   // enforce unit normal
```

Comment:
I think this piece of code will throw an error if all 3 normal args are variables?
B/c length or current normal = zero?